p2p/host/hostv2/hostv2.go

```diff
@@ -54,16 +85,38 @@ type HostV2 struct {
 	logger *zerolog.Logger
 }
 
+func (host *HostV2) getTopic(topic string) (topicHandle, error) {
+	host.lock.Lock()
+	defer host.lock.Unlock()
+	if t, ok := host.joined[topic]; ok {
+		host.logger.Info().Str("topic", topic).Msg("Found Topic")
+		return t, nil
+	} else if t, err := host.joiner.JoinTopic(topic); err != nil {
+		return nil, errors.Wrapf(err, "cannot join pubsub topic %x", topic)
+	} else {
+		host.joined[topic] = t
+		host.logger.Info().Str("topic", topic).Msg("Joined Topic")
+		return t, nil
+	}
+}
+
 // SendMessageToGroups sends a message to one or more multicast groups.
-func (host *HostV2) SendMessageToGroups(groups []nodeconfig.GroupID, msg []byte) error {
-	var error error
+// It returns a nil error if and only if it has succeeded to schedule the given
+// message for sending.
+func (host *HostV2) SendMessageToGroups(groups []nodeconfig.GroupID, msg []byte) (err error) {
 	for _, group := range groups {
-		err := host.pubsub.Publish(string(group), msg)
-		if err != nil {
-			error = err
+		t, e := host.getTopic(string(group))
+		if e != nil {
+			err = e
```

Comment:
Should log each of the error separately right?

Reply:
the error will be surface to the caller, which is sendMesagewithRetry/WithoutRetry, we will still see the error from the top level caller.